### PR TITLE
Feat/flocking

### PR DESCRIPTION
**Pull request by 0x6c23**
I added the NavQuadrantSystem so we only steer away from neighbors in our quadrant. I also renamed the NavLerpSystem to NavSteeringSystem and added a NavSteering component. The NavSteeringSystem gets the steering vectors from the FlockingSystem and calculates a new heading for the entity.

The new NavMoveSystem will take this heading and apply it to the entity, aswell as calculate an appropriate rotation for the entity, so it will face the direction of movement with all steering behaviors included. I also added FlockingSettings to the NavSystem and additional settings to the NavAgent. The sequence of the jobs was changed so they work with the new systems.

Oh, I added a gif, as well.

There might be errors/ flaws in my code, please highlight them to me. 

**Yet to be implemented:**

The steering behaviors can push entities into walls, might add some raycasting to prevent that if that sounds like a good idea?

Also, I suggest rewriting the follow system so it provides an additional (**weighted**) steering vector for the NavSteeringSystem. So it would be more flexible.

### CODE REVIEW

File: Packages/com.reese.nav/Agent/Systems/NavFollowSystem.cs
Comment (reeseschultz) [inline Packages/com.reese.nav/Agent/Systems/NavFollowSystem.cs]: I think this import may have been inadvertently added during debugging. No biggie, just letting you know.

File: Packages/com.reese.nav/Agent/Systems/NavMoveSystem.cs
Comment (reeseschultz) [inline Packages/com.reese.nav/Agent/Systems/NavMoveSystem.cs:42]: This general area could use some newlines between code blocks for readability.

File: Packages/com.reese.nav/Flocking/NavFlockingSystem.cs
Comment (reeseschultz) [inline Packages/com.reese.nav/Flocking/NavFlockingSystem.cs]: I would rename `navSettings` to `settings`.

File: Packages/com.reese.nav/Flocking/NavFlockingSystem.cs
Comment (reeseschultz) [inline Packages/com.reese.nav/Flocking/NavFlockingSystem.cs]: I would rename `navFollowFromEntity` to `followFromEntity`.

File: Packages/com.reese.nav/Flocking/NavFlockingSystem.cs
Comment (reeseschultz) [inline Packages/com.reese.nav/Flocking/NavFlockingSystem.cs]: I would rename `navSteering` to `steering`.

File: Packages/com.reese.nav/Flocking/NavFlockingSystem.cs
Comment (reeseschultz) [inline Packages/com.reese.nav/Flocking/NavFlockingSystem.cs:57]: This line could use a newline underneath it.

File: Packages/com.reese.nav/Quadrant/NavQuadrantSystem.cs
Comment (reeseschultz) [inline Packages/com.reese.nav/Quadrant/NavQuadrantSystem.cs]: I'm pretty sure the super class' `OnCreate()` and `OnDestroy()` are empty, but you can leave the calls to them here if you want.

File: Packages/com.reese.nav/Quadrant/NavQuadrantSystem.cs
Comment (reeseschultz) [inline Packages/com.reese.nav/Quadrant/NavQuadrantSystem.cs:23]: I don't like public statics but, honestly, even if we encapsulated this, someone could still abuse it. Fire burns when you touch it, lol. I think this is fine.

File: Packages/com.reese.nav/Quadrant/NavQuadrantSystem.cs
Comment (reeseschultz) [inline Packages/com.reese.nav/Quadrant/NavQuadrantSystem.cs]: The `private` keyword is extraneous. Other than that, brilliant work in this class. I love it. Good job.

File: Packages/com.reese.nav/Quadrant/NavQuadrantSystem.cs
Comment (reeseschultz) [inline Packages/com.reese.nav/Quadrant/NavQuadrantSystem.cs:72]: You could convert this into an arrow function if you want, but you don't have to.

File: README.md
Comment (reeseschultz) [inline README.md]: This gif kicks ass. You can remove the other (previous) gif if you want, because it's taking up too much space.

I'll eventually do a new showcase gif.

**Review by reeseschultz — CHANGES_REQUESTED** [review]
I noted some minor changes, but overall this is awesome. Your code looks wonderful. I don't see anything obviously wrong.

You're welcome to add extra raycasting to handle walls, but you don't have to. Up to you. As for rewriting the steering system, you're welcome to rewrite it as well so it's more flexible. Otherwise, it can always be updated in the future. Also, the nav README needs to be updated for your changes (I don't think you updated it), but if you don't, I can just update it after we merge. Not a big deal.

Despite the minor changes needed, this is essentially tentative approval. Tomorrow morning, I'll run the code and try to ensure that we don't miss anything that could cause problems.

**Comment by 0x6c23** [comment]
Alright, I pushed the requested changes. Will open a new pull request for the extra raycasting + steering elements. Thanks for your comments!

**Review by reeseschultz — APPROVED** [review]
Excellent work. And thank you for updating the documentation!

**Pull request merged**
